crates/ra_hir_ty/src/tests/traits.rs

```diff
@@ -540,6 +540,42 @@ mod ops {
     );
 }
 
+#[test]
+fn infer_ops_index_int() {
+    check_types(
+        r#"
+//- /main.rs crate:main deps:std
+struct Bar;
+struct Foo;
+
+impl std::ops::Index<u32> for Bar {
+    type Output = Foo;
+}
+
+struct Range;
+impl std::ops::Index<Range> for Bar {
+    type Output = Bar;
+}
+
+fn test() {
+    let a = Bar;
+    let b = a[1];
+    b;
+  //^ Foo
+}
+
+//- /std.rs crate:std
+#[prelude_import] use ops::*;
+mod ops {
+    #[lang = "index"]
+    pub trait Index<Idx> {
+        type Output;
+    }
+}
```

Comment:
at some point we need to figure a better way to specify such preludes...

I guess, we can do `//- core`, and have that auto-magically insert everything, but I wonder if it makes sense to keep stuff minimal? 

LGTM!

Reply:
I do kind of like having it explicit what types/traits are actually needed by the test. Also, keeping it small helps when debugging tests.

Reply:
Yeah, that's true. But we might do  something granular, but still with only one `trait Index` written down

```rust
//- include(std::core::Index, std::result::Result)
```

but that won't you see at a glance what's going on...

Reply:
That might be nice. Maybe there should just be std/core "snippets" that the tests can include.